### ISSUE TITLE
Linq implementation generates wrong query for an ORDER BY following a SELECT

### ISSUE THREAD

**Issue by brokentriangles** [Azure/azure-cosmos-dotnet-v2#246]
There appears to be a bug in the Linq implementation when generating a query containing an ORDER BY following a SELECT.

```
class Container<T>
{
    public System.Guid id { get; set; }
    public string Key { get; set; }
    public T Content { get; set; }
}

public class Content
{
    public string Version { get; set; }
}

var content = client.CreateDocumentQuery<Container<Content>>(collection.DocumentsLink)
    .Where(container => container.Key == "MyKey")
    .Select(container => container.Content)
    .OrderBy(content => content.Version);
```

This generates the following query, returning zero results.
`SELECT VALUE root["Content"] FROM root WHERE (root["Key"] = "MyKey") ORDER BY root["Version"] ASC`

The following query returns results as expected.
`SELECT VALUE root["Content"] FROM root WHERE (root["Key"] = "MyKey") ORDER BY root["Content"]["Version"] ASC`

**Comment by bpatra** [comment]
It may be related to issue #208

**Comment by kirankumarkolli** [comment]
It repros with 1.17.0 version as well.

**Comment by wtodd** [comment]
Any updates on this?

**Comment by frankabbruzzese** [comment]
Bug persists also in 1.7.1! Any plan to fix this in future versions?

**Comment by jwyoung9** [comment]
Also having this issue.  Any update?

**Comment by bytelabsco** [comment]
Still broken in Microsoft.Azure.DocumentDB.Core v1.9.1

**Comment by kirankumarkolli** [comment]
@bchong95 please look into this.

**Comment by khdang** [comment]
I believe this has been fixed in 2.1.3 release. Please reopen if it's still affecting any scenarios.
